api/src/controllers/website.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * This module contains handlers for the "clientWebsite" route.
+ * @module website
+ */
+const axios = require('axios');
+
+const Website = require('../models/website.model');
+
+const { fillErrorObject } = require('../middleware/error');
+
+var ObjectId = require('mongoose').Types.ObjectId;
+
+/**
+ * Handle GET request from /clientWebsite/:team_id
+ * @param {*} req request object, containing teamId
+ * @param {*} res response object
+ * @returns 200: the team's website info was found
+ * @returns 404: team's website info is not found
+ * @returns 500: server error
+ */
+async function getWebPageDetails(req, res) {
+  try {
+    const websiteInfo = await Website.findOne({
+      teamId: new ObjectId(req.params.team_id),
+    });
+    if (!websiteInfo) {
+      return res
+        .status(404)
+        .send(
+          'No Website info found for this Team (This is likely because they have not added any website information yet)'
+        );
+    }
+    return res.status(200).send(websiteInfo);
+  } catch (error) {
+    return res.status(500).json(`Server Error: ${error.message}`);
+  }
+}
+
+/**
+ * Initializes info about the website.
+ * @param info: info to initialize schema with, must include teamId
+ * @returns 200: Initial website Info successfully added to DB
+ * @returns 400: Team Id given already has a website created
+ */
+async function createInitialWebsiteInfo(info) {
+  Website.findOne({ teamId: info.teamId })
+    .then((website) => {
+      if (website == null) {
+        // Create Website data
+        Website.create(info).catch((err) => {
+          throw err;
+        });
+      }
+    })
+    .catch((err) => {
```

Comment:
Is there a reason we're using `.then().catch()` i.e. promise style here, but async-await style elsewhere? Maybe we should stick to one convention

Reply:
Fair point, I've converted majority into promise-style as this seems to be more common in other files as well.